### PR TITLE
Cache AgentVersion and PeerData in network globals

### PR DESCRIPTION
**Pull request by dapplion**
**Motivation**

Currently we read AgentVersion and ClientKind from the peerstore everytime it's needed. In the libp2p bump https://github.com/ChainSafe/lodestar/pull/3661 this can trigger disk reads. This data is not necessary to persist besides the lifetime of the peer.

This PR extends https://github.com/ChainSafe/lodestar/pull/3838 adding agentVersion and clientKind to the ephemeral data attached to connected peers, which is dropped on disconnection.

In the future we should do a PR upstream to libp2p to emit the response of the identity protocol to prevent having to poll agentVersions from the peerstore

**Description**

- Cache AgentVersion and PeerData in network globals
- Move encodingPreference to PeerData

### CODE REVIEW

**Comment by wemeetagain** [comment]
There's an event emitted from the peerstore.
`'change:metadata', {peerId, metadata}: {peerId: PeerId; metadata: Map<string, Uint8Array>}`

**Review by twoeths — DISMISSED** [review]

File: packages/lodestar/src/network/globals.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+/**
+ * Make data available to multiple components in the network stack.
+ * Due to class dependencies some modules have circular dependencies, like PeerManager - ReqResp.
+ * This third party class allows data to be available to both.
+ *
+ * The pruning and bounding of this class is handled by the PeerManager
+ */
+export class NetworkGlobals {
```
Comment (wemeetagain) [inline packages/lodestar/src/network/globals.ts]: Maybe this would be better called `PeerMetadata` and `peerMetadata` throughout.

Thoughts?

Comment (twoeths) [reply packages/lodestar/src/network/globals.ts]: we have consensus MetaData already, so maybe make this one `PeersData`? there is internal `PeerData` inside

File: packages/lodestar/src/network/peers/peerManager.ts [hunk 2]
```diff
@@ -553,15 +567,20 @@ export class PeerManager {
   /** Register peer count metrics */
   private async runPeerCountMetrics(metrics: IMetrics): Promise<void> {
     let total = 0;
+
+    // TODO: Should we use here `this.connectedPeers`?
```
Comment (wemeetagain) [inline packages/lodestar/src/network/peers/peerManager.ts]: I don't think we need/want to use `this.connectedPeers` here?

File: packages/lodestar/src/network/globals.ts (hunk 1 above)
Comment (wemeetagain) [reply packages/lodestar/src/network/globals.ts]: would still like to see the name changed to be more specific than "network globals"

**Review by wemeetagain — APPROVED** [review]
LGTM

**Pull request merged**